src/ffmpeg.cc

```diff
@@ -51,52 +58,79 @@ FFmpeg::~FFmpeg() {
 	wait();
 	// TODO: use RAII for freeing resources (to prevent memory leaks)
 	QMutexLocker l(&s_avcodec_mutex); // avcodec_close is not thread-safe
-	if (pAudioCodecCtx) avcodec_close(pAudioCodecCtx);
-	if (pFormatCtx) avformat_close_input(&pFormatCtx);
 	}
 
 double FFmpeg::duration() const {
 	double d = m_running ? pFormatCtx->duration / double(AV_TIME_BASE) : getNaN();
 	return d >= 0.0 ? d : getInf();
 }
 
-// FFMPEG has fluctuating API
-#if LIBAVCODEC_VERSION_INT < ((52<<16)+(64<<8)+0)
-#define AVMEDIA_TYPE_VIDEO CODEC_TYPE_VIDEO
-#define AVMEDIA_TYPE_AUDIO CODEC_TYPE_AUDIO
-#endif
-
 void FFmpeg::open() {
 	QMutexLocker l(&s_avcodec_mutex);
+#if LIBAVFORMAT_VERSION_INT < AV_VERSION_INT(58,9,100) // Got deprecated in 4.0 but needed for earlier versions
 	av_register_all();
+#endif
 	av_log_set_level(AV_LOG_ERROR);
-	if (avformat_open_input(&pFormatCtx, m_filename.c_str(), NULL, NULL)) throw std::runtime_error("Cannot open input file");
-	if (avformat_find_stream_info(pFormatCtx, NULL) < 0) throw std::runtime_error("Cannot find stream information");
+
+	AVFormatContext *ps{};
+
+	if (auto err = avformat_open_input(&ps, m_filename.c_str(), NULL, NULL); err != 0)
```

Comment:
this uses `C++17 feature called init-statement in if/switch` which is not enabled per default on msvc (windows)
For now let's move the variable init to a different line. Let's tackle windows builds in a different pr